packages/block/src/header.ts

```diff
@@ -361,20 +419,35 @@ export class BlockHeader {
 
   /**
    * Returns the block header in JSON format
-   *
-   * @see {@link https://github.com/ethereumjs/ethereumjs-util/blob/master/docs/index.md#defineproperties|ethereumjs-util}
    */
   toJSON(_labels: boolean = false): { [key: string]: string } | string[] {
     // Note: This never gets executed, defineProperties overwrites it.
-    return {}
+    /*return Object.create({
+      parentHash: this.parentHash,
+      uncleHash: this.uncleHash,
+      coinbase: this.coinbase,
+      stateRoot: this.stateRoot,
+      transactionsTrie: this.transactionsTrie,
+      receiptTrie: this.receiptTrie,
+      bloom: this.bloom,
+      difficulty: this.difficulty,
+      number: this.number,
+      gasLimit: this.gasLimit,
+      gasUsed: this.gasUsed,
+      timestamp: this.timestamp,
+      extraData: this.extraData,
+      mixHash: this.mixHash,
+      nonce: this.nonce,
+    })*/
+    return {} //TODO: FIXME
   }
 
   private _getHardfork(): string {
     const commonHardFork = this._common.hardfork()
 
     return commonHardFork !== null
       ? commonHardFork
-      : this._common.activeHardfork(bufferToInt(this.number))
+      : this._common.activeHardfork(this.number.toNumber())
```

Comment:
Also here; Common should support BNs.

Reply:
see https://github.com/ethereumjs/ethereumjs-vm/pull/883#discussion_r501315274